### ISSUE TITLE
Extension of non-square matrix with all negative values in lapjv

### ISSUE THREAD

**Issue by jvlmdr** [gatagat/lap#21]
Hey, I think there is a bug in the code for extending a non-square matrix to a square matrix.

Consider the all-negative cost matrix
```
```

**Comment by gatagat** [comment]
Hi, the cost matrix must be non-negative.

Do you see the error with cost + 30?

**Comment by jvlmdr** [comment]
Oh ok, sorry, I didn't realise :)

**Comment by jvlmdr** [comment]
And yep, it gives the correct solution for the original cost matrix.

**Comment by gatagat** [comment]
This should definitely be mentioned in the docs. If you have some time, please, send a PR.

**Comment by jvlmdr** [comment]
Maybe it would be better to also raise an exception if one of the costs is negative, or subtract the minimum? But are you sure it doesn't work with negative costs? It seems to give the correct result.

**Comment by gatagat** [comment]
@jvlmdr You are right, my bad, lack of sleep, lack of time... It is the dual where the reduced costs have to be non-negative, so no issue in the algorithm itself, the extension is wrong.

**Comment by gatagat** [comment]
And your patch #22 is sound. The only question that remains is what has a better runtime: extend the way you do, or shift the costs to zero first and extend as before.
